src/test/java/com/example/dog/dao/JdbcDogDAOTest.java

```diff
@@ -0,0 +1,62 @@
+package com.example.dog.dao;
+
+import com.example.dog.model.Dog;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@ContextConfiguration(locations={"classpath:/controllers-context.xml"})
+public class JdbcDogDAOTest extends AbstractTestNGSpringContextTests {
+
+    @Autowired
+    private JdbcDogDAO jdbcDogDao;
+
+    @Test
+    public void possibleToSaveNameWithMaxLength() {
```

Comment:
All fields - not just names. You can use Integer.MAX_INT, etc, to generate max values for all the fields.